src/vm/functions/special.rs

```diff
@@ -200,6 +224,54 @@ fn handle_pox_v2_api_contract_call(
                 return Ok(());
             }
         }
+    } else if function_name == "stack-extend" || function_name == "delegate-stack-extend" {
+        // applying a pox lock at this point is equivalent to evaluating a transfer
+        debug!(
+            "Handle special-case contract-call to {:?} {} (which returned {:?})",
+            boot_code_id("pox-2", global_context.mainnet),
+            function_name,
+            value
+        );
+
+        runtime_cost(
+            ClarityCostFunction::StxTransfer,
+            &mut global_context.cost_track,
+            1,
+        )?;
+
+        if let Ok((stacker, unlock_height)) = parse_pox_extend_result(value) {
+            match StacksChainState::pox_lock_extend_v2(
+                &mut global_context.database,
+                &stacker,
+                unlock_height as u64,
+            ) {
+                Ok(locked_amount) => {
+                    if let Some(batch) = global_context.event_batches.last_mut() {
+                        batch.events.push(StacksTransactionEvent::STXEvent(
+                            STXEventType::STXLockEvent(STXLockEventData {
+                                locked_amount,
+                                unlock_height,
+                                locked_address: stacker,
+                            }),
+                        ));
+                    }
+                }
+                Err(ChainstateError::DefunctPoxContract) => {
+                    return Err(Error::Runtime(RuntimeErrorType::DefunctPoxContract, None))
+                }
+                Err(e) => {
+                    panic!(
+                        "FATAL: failed to extend lock from {} until {}: '{:?}'",
+                        stacker, unlock_height, &e
+                    );
+                }
+            }
+
+            return Ok(());
+        } else {
+            // nothing to do -- the function failed
```

Comment:
why does the error get suppressed? please add to comment

Reply:
This is the case where `stack-extend` or `delegate-stack-extend` return an `(err ...)` response.  No additional work needs to happen in this case.